lib/manager/pip_requirements/extract.ts

```diff
@@ -62,13 +62,12 @@ export function extractPackageFile(
       if (isSkipComment(comment)) {
         dep.skipReason = SkipReason.Ignored;
       }
-      regex.lastIndex = 0;
-      const matches = regex.exec(line.split(' \\')[0]);
+      const matches = pkgValRegex.exec(line.split(' \\')[0]);
```

Comment:
Why we split here? Isn't the regex already capture only the dep and value?

Reply:
It's to exclude when there are hashes in the line. Doing this way is easier than regex. I'll add a test for it.